include/swift/AST/Expr.h

```diff
@@ -1435,24 +1435,33 @@ class TypeExpr : public Expr {
 
 class TypeValueExpr : public Expr {
   GenericTypeParamDecl *paramDecl;
+  TypeRepr *repr;
```

Comment:
You shouldn't need a paramDecl if you're going to add a TypeRepr. An unqualified reference to a value generic parameter can also be represented by a TypeRepr.